src/watchdog.js

```diff
@@ -214,15 +233,28 @@ export default class Watchdog {
 
 				this._lastDocumentVersion = editor.model.document.version;
 				this._data = editor.data.get();
+				this.state = 'ready';
 			} );
 	}
 
 	/**
-	 * Destroys the current editor instance by using the destructor passed to the {@link #setDestructor `setDestructor()`} method.
+	 * Destroys the current editor instance by using the destructor passed to the {@link #setDestructor `setDestructor()`} method
+	 * and set state to `destroyed`.
 	 *
 	 * @returns {Promise}
 	 */
 	destroy() {
+		return this._destroy().then( () => {
+			this.state = 'destroyed';
```

Comment:
I think you should set the state before, not after you destroy the editor. Note that as soon as `destroy()` method is called one should thread the editor/watchdog as destroyed. Also, this is how it works in the editor.

Reply:
Ok, I've changed it.